syntax/go.vim

```diff
@@ -162,15 +162,17 @@ endif
 syn match       goSingleDecl        /\%(import\|var\|const\) [^(]\@=/ contains=goImport,goVar,goConst
 
 " Integers
-syn match       goDecimalInt        "\<-\=\(0\|[1-9]\d*\)\%([Ee][-+]\=\d\+\)\=\>"
-syn match       goHexadecimalInt    "\<-\=0[xX]\x\+\>"
-syn match       goHexadecimalError  "\<-\=0[xX]\x*[^ \t0-9A-Fa-f]\S*\>"
-syn match       goOctalInt          "\<-\=0\o\+\>"
-syn match       goOctalError        "\<-\=0[XxBb\]]\@!\o*[^ \t0-7]\S*\>"
-syn match       goBinaryInt         "\<-\=0[bB][01]\+\>"
-syn match       goBinaryError       "\<-\=0[bB][01]*[^ \t01]\S*\>"
+syn match       goDecimalInt        "\<-\=\(0\|[1-9]_\?\(\d\+_\?\)*\)\%([Ee][-+]\=\d\+\)\=\>"
+syn match       goDecimalErrorInt   "\<-\=\(_\(\d\+_*\)\+\|\(\d\+_*\)\+_\{2,\}\(\d\+_*\)\+\|\(\d\+_*\)\+_\+\)\%([Ee][-+]\=\d\+\)\=\>"
```

Comment:
To avoid matching the octal errors, the second and third branches of this should have `[1-9]` as their starting character instead of `\d`.

Reply:
Yeah, it makes sense. Fixed.